packages/apollo-codegen-swift/src/language.ts

```diff
@@ -311,12 +311,33 @@ export class SwiftGenerator<Context> extends CodeGenerator<
     super(context);
   }
 
-  multilineString(string: string) {
+  /**
+   * Outputs a multi-line string
+   *
+   * @param string - The Multi-lined string to output
+   * @param suppressMultilineStringLiterals - If true, will output the multiline string as a single trimmed
+   *                                          string to save bandwidth.
+   *                                          NOTE: This preference will be ignored if your GraphQL query
+   *                                          contains a triple-quote GraphQL string literal, since that's
+   *                                          the same as Swift's string literal and will break the query.
+   */
+  multilineString(string: string, suppressMultilineStringLiterals: Boolean) {
     // Disable trimming if the string contains """ as this means we're probably printing an
     // operation definition where trimming is destructive.
-    this.printOnNewline(
-      SwiftSource.string(string, /* trim */ !string.includes('"""'))
-    );
+    if (string.includes('"""')) {
+      this.printOnNewline(SwiftSource.string(string, /* trim */ false));
+    } else {
+      if (suppressMultilineStringLiterals) {
+        this.printOnNewline(SwiftSource.string(string, /* trim */ true));
+      } else {
+        // Use a multiline string literal
+        this.printOnNewline(SwiftSource.raw`"""`);
+        string.split("\n").forEach(line => {
+          this.printOnNewline(SwiftSource.raw`${line}`);
```

Comment:
This is wrong; if the line contains a backslash, it won't be escaped. We need a modified form of `SwiftSource.string` that omits the surrounding quotes (maybe just called something like `SwiftSource.stringContents`). Doing that directly will also escape double quotes, which is unnecessary (as long as there aren't 3 in a row), so perhaps we should instead have a specific `SwiftSource.multilineString` that emits the triple quotes and does the modified escaping.

Reply:
Or instead of escaping we can do a raw string literal instead, we just have to calculate the appropriate string literal prefix (not that we should run into a conflict with Apollo sources, but it's better to be correct instead of assuming the string `"""#` will never show up in an Apollo query).

Reply:
I took a look at using raw strings, but from what I saw raw strings and multiline strings do NOT play well together.

Reply:
Yeah they do.

```swift
#"""
This is a multiline string
It can include """
It just can't put a hash after it.
Escapes look like \#t.
"""#
```

(Ignore GitHub's syntax highlighting, that's not correct).

My only worry about using raw strings is the logic to calculate the appropriate `#` count. I think it's simpler just to use a regular multiline string and add the appropriate escaping as needed.

Reply:
I'll take a look at a test around a line with a backslash

Reply:
Aha, i was fucking something up with the raw strings. I think I'd probably just use `###` as the delimiter by default so we don't have to count `#`s. Plus raw strings mean we don't have to screw around with slashies too

Reply:
I'm not actually sure it's legal to get `"""#` or `\#` in a GraphQL query, so it's possible we don't have to worry. I'm currently trying to figure out if it's possible to construct something that parses like this.

I am assuming comments get stripped, otherwise anything goes in a comment.

Reply:
Ok no, it's possible to get the token `\"""#` to show up in a GraphQL multiline string.

```graphql
query Bar {
  foo(str: """wat\"""#""") {
    id
  }
}
```

I can also apparently pass `\#` through the GraphQL parse step, though that's not a known escape so I assume it should fail at runtime.

Reply:
The best thing to do here is probably to just add basic string escaping here and then I'll try and do a followup PR that cleans it up such that it does the "optimal" formatting.